hilti/toolchain/include/compiler/plugin.h

```diff
@@ -101,7 +100,7 @@ struct Plugin {
      * @param arg3 file associated with the input stream
      * @return module AST if parsing succeeded
      */
-    Hook<Result<declaration::Module*>, hilti::Builder*, std::istream&, hilti::rt::filesystem::path> parse;
+    Hook<Result<declaration::Module*>, hilti::Builder*, std::istream&, const hilti::rt::filesystem::path&> parse;
```

Comment:
Interesting, looks like due to automatic arg type conversions for `std::function` a function pointer might be _more type-safe_ 🤯

Reply:
Yeah, I was thinking the same!